File: ContainsDuplicate.cs
using System.Collections;

public class Solution
{
    //O(n)
    //O(n)
    public bool ContainsDuplicate(int[] nums)
    {
        if (nums.Length <= 1) return false;
        var hashSet = new HashSet<int>();

        foreach (var item in nums)
        {
            if (hashSet.Contains(item))
            {
                return true;
            }
            else
            {
                hashSet.Add(item);
            }
        }
        
        return false;
    }
}

File: Go/containsDuplicate.go
package main

//O(N)
//O(N)
func containsDuplicate(nums []int) bool {

	if len(nums) <= 0 {
		return false
	}

	var set = make(map[int]bool)

	for _, num := range nums {
		if set[num] {
			return true
		}
		set[num] = true
	}

	return false
}
